File: BestTimeToBuyAndSellStock/BestTime.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        if len(prices) == 0: return 0
        max_profit = 0
        min_price = prices[0]
        for i in range(len(prices)):
                profit = prices[i] - min_price
                max_profit = max(profit, max_profit)
                min_price = min(min_price, prices[i])
        return max_profit

File: BestTimeToBuyAndSellStock/BestTime.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function (prices) {
  if (prices.length === 0) return 0;
  let max_profit = 0;
  let min_price = prices[0];
  for (let i = 0; i < prices.length; i++) {
    profit = prices[i] - min_price;
    max_profit = Math.max(profit, max_profit);
    min_price = Math.min(min_price, prices[i]);
  }
  return max_profit;
};
